overlord/state/taskrunner.go

```diff
@@ -28,6 +28,17 @@ import (
 	"github.com/snapcore/snapd/logger"
 )
 
+// HookFunc is the type of function that can be registered as a hook.
+type HookFunc func()
+
+// HookEvent are types of different hooks that can be registered handlers for.
+type HookEvent int
+
+const (
+	// TaskExhaustionHook is executed when there are no more tasks to be run.
+	TaskExhaustionHook HookEvent = iota
```

Comment:
as it's an event it shouldn't probably have Hook in the name, just TaskExhaustion?

Reply:
Yea was a bit confused on this, the input here is greatly appreciated.